### ISSUE TITLE
Children() method run slowly

### ISSUE THREAD

**Issue by P-A-C** [OWASP/rbac#112]
Children() method takes a lot of time to return the child roles, is this normal?
I specify, I handle a lot of roles.
If it's normal, how do I get around it?

**Comment by abiusx** [comment]
It is normal, and children() should not be called regularly. It is only meant for a UI page listing the roles, not for permission checks.
What are you using it for?

**Comment by P-A-C** [comment]
Of course, I understand that this method should not be misused, but I waited almost 15 seconds to start listing and displaying the roles.
Is there a possible system to reduce this time?

**Comment by abiusx** [comment]
That's not right.
We're using Nested Sets solely to reduce this load. It definitely shouldn't take 15 seconds.
Did you change any of the database indexes? Can you let me know the cardinality of tables?

**Comment by P-A-C** [comment]
no, I assure you that I didn't changed anything on the database.
I only used the implemented methods and did my tree.

**Comment by abiusx** [comment]
How many roles/permissions are there?

**Comment by P-A-C** [comment]
almost a thousand

**Comment by abiusx** [comment]
There should be no issues. Can you share your database with me for testing? 
There is definitely something wrong.
I have tested the library with upwards of 1 million rows and results are under 1 sec.

**Comment by P-A-C** [comment]
of course here it is.

[rbac.zip](https://github.com/OWASP/rbac/files/2639794/rbac.zip)

**Comment by abiusx** [comment]
I was unable to open the zip file.

**Comment by P-A-C** [comment]
sorry
[rbac.txt](https://github.com/OWASP/rbac/files/2639803/rbac.txt)

**Comment by P-A-C** [comment]
and I do my test of children methode on role_id == 647 (ROLE_MANAGER/1) an dit takes a lot of time

**Comment by abiusx** [comment]
I see two issues with your database.
1. There are some 15,000 permissions and 1,000 roles, and about 50,000 role-permission relations.
However, that doesn't seem to be the real issue:

2. Your permissions have slashes in their name.
If you want to use the path based query, you should probably not use slashes in your role/permission names, as they will conflict with the pathing.

**Comment by P-A-C** [comment]
Yes, I will try to reduce my roles, permissions and relationships. I will also replace the slashes with another character.
If the problem persists, I will come back to this topic.
Thank you so much.

**Comment by abiusx** [comment]
The number of roles and permissions shouldn't really be an issue.
Just try to replace all slashes with _ and see if it gets fixed.

**Comment by P-A-C** [comment]
Okay.
thank you for being so available
